src/servers/Server_Zone/Script/ScriptManager.cpp

```diff
@@ -42,24 +52,65 @@ bool Core::Scripting::ScriptManager::init()
    loadDir( g_serverZone.getConfig()->getValue< std::string >( "Settings.General.Scripts.Path", "./compiledscripts/" ),
             files, m_nativeScriptManager->getModuleExtension() );
 
+   uint32_t scriptsFound = 0;
+   uint32_t scriptsLoaded = 0;
+
    for( auto itr = files.begin(); itr != files.end(); ++itr )
    {
       auto& path = *itr;
 
-      m_nativeScriptManager->loadScript( path );
+      scriptsFound++;
+
+      if( m_nativeScriptManager->loadScript( path ) )
+         scriptsLoaded++;
    }
 
+   g_log.info( "ScriptManager: Loaded " + std::to_string( scriptsLoaded ) + "/" + std::to_string( scriptsFound ) + " scripts successfully" );
+
+   watchDirectories();
+
    return true;
 }
 
+void Core::Scripting::ScriptManager::watchDirectories()
+{
+   Watchdog::watchMany( g_serverZone.getConfig()->getValue< std::string >( "Settings.General.Scripts.Path", "./compiledscripts/" ) + "*" + m_nativeScriptManager->getModuleExtension(),
+   [ this ]( const std::vector< ci::fs::path >& paths )
+   {
+      if( !m_firstScriptChangeNotificiation )
+      {
+         // for whatever reason, the first time this runs, it detects every file as changed
+         // so we're always going to ignore the first notification
+         m_firstScriptChangeNotificiation = true;
+         return;
+      }
+
+      for( auto path : paths )
+      {
+         if( m_nativeScriptManager->isModuleLoaded( path.stem().string() ) )
+         {
+            g_log.debug( "Reloading changed script: " + path.stem().string() );
+
+            m_nativeScriptManager->reloadScript( path.stem().string() );
+         }
+         else
+         {
+            g_log.debug( "Loading new script: " + path.stem().string() );
+
+            m_nativeScriptManager->loadScript( path.string() );
+         }
+      }
+   });
+}
+
 void Core::Scripting::ScriptManager::loadDir( std::string dirname, std::set<std::string>& files, std::string ext )
 {
 
    g_log.info( "ScriptEngine: loading scripts from " + dirname );
 
    boost::filesystem::path targetDir( dirname );
 
-   boost::filesystem::recursive_directory_iterator iter( targetDir ), eod;
+   boost::filesystem::directory_iterator iter( targetDir ), eod;
```

Comment:
We do not do multiple declaration per line, this is not documented anywhere, should be though ;)